engine/jobmaster/dm/metrics.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dm
+
+import (
+	"github.com/pingcap/tiflow/engine/pkg/promutil"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	jobFactory        = promutil.NewFactory4Framework()
```

Comment:
Note that `promutil.NewFactory4Framework()`  can only be used inside the framework.
Should `jm.metricFactory ()` be used to create the monitor item here?

Reply:
I tried to use `jm.MetricFactory()` to create, but I failed. It seems that using `metricFactory()` function need a timing that after some initialization. Are you sure we can use `jm.MetricFactory()` in `NewTaskManager`?